### ISSUE TITLE
Doc enhancement: Must update AWS ECR tokens w/remote DB and kube cluster redeployment

### ISSUE THREAD

**Issue by cmcconnell1** [teamhephy/workflow#93]
Hello All,

Not sure where this should go, but wanted to offer this back to other deis workflow users.  
I ran into this issue over the weekend that I had never seen before wherein we had a very old kube cluster redeployment/replacement and wanted to share this to hopefully help others.

Notes:
- We are using remote postgreSQL database for deis database/state.
- Using the latest hemphy deis helm chart installing with custom values.yaml file.
- Previous cluster redeployments did not require this update--since we have remote database state, other kube cluster redeployments just required installing deis workflow into the fresh, updated kube cluster with the values.yaml confiuguring remote database storage and deis would pickup where it left off--pulling all specified docker images from our remote AWS ECR repo.
- This issue is relevant for the forked team hemphy dies workflow--i.e. previously when using the original deis workflow chart, this was not necessary, so it through me for a bit of a loop...

deis controller logs havent seen this before 
```sh
alias login-deis-stage='deis login http://deis.stage.foo.com --username=fluffykitty --password=foo'
login-deis-stage && $(aws ecr get-login --no-include-email --region us-west-1 --profile my-admin-profile) && deis pull myrepo.dkr.ecr.us-west-1.amazonaws.com/app1/my-app:0.8.134 -a my-deis-app
```

- results in:
```sh
INFO Logging into Registry myrepo.dkr.ecr.us-west-1.amazonaws.com/foo/myapp with username AWS
ERROR [aergo]: (app::deploy): 500 Server Error: Internal Server Error ("b'{"message":"login attempt to https://myrepo.dkr.ecr.us-west-1.amazonaws.com/v2/ failed with status: 403 Forbidden"}'")
Traceback (most recent call last):
 File "/usr/local/lib/python3.5/dist-packages/docker/client.py", line 170, in _raise_for_status
  response.raise_for_status()
 File "/usr/local/lib/python3.5/dist-packages/requests/models.py", line 940, in raise_for_status
  raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.38/auth
...
```

The above exception was the direct cause of the following exception:
​
```sh
Traceback (most recent call last):
 File "/app/api/models/app.py", line 531, in deploy
  release.check_image_access()
 File "/app/api/models/release.py", line 144, in check_image_access
  raise DeisException(str(e)) from e
api.exceptions.DeisException: 500 Server Error: Internal Server Error ("b'{"message":"login attempt to https://myrepo.dkr.ecr.us-west-1.amazonaws.com/v2/ failed with status: 403 Forbidden"}'")
```

#### Solution 
- Reset AWS ECR token with new updated value and update the remote database with that value via `deis registry:set` for all deis apps.
  - get updated ecr token
  ```sh
  aws ecr get-login --no-include-email --region us-west-1 --profile my-admin-profile
  ```

  - Update ecr token for all deis apps--i.e. for i in all_deis_apps; do
  ```sh
  deis registry:set -a $i username=AWS password=XXXXXXXX123123123123122312132
  ```

##### Validate/profit
Now the fresh hemphy/deis workflow install on the new kube cluster (using the off cluster persistent data from the previous deis workflow--stored in the remote database) works as expected.  
  - I.e.: deis now pulls all specified docker images per the remote database settings from the previous kube cluster and deis workflow etc.

**Comment by kingdonb** [comment]
```
No results found for site:docs.teamhephy.info "registry:set"
```
https://www.google.com/search?q=site%3Adocs.teamhephy.info+%22registry%3Aset%22

Hmm this seems to be a feature that was documented on Deis v1, but not anymore

**Comment by Cryptophobia** [comment]
Hi @cmcconnell1 , this behavior was reverted back to what was the original intention in https://github.com/teamhephy/controller/pull/103. We had several users complaining so we reverted the changes on the controller api.

The controller should now try the main authentication registry-secret for all apps so users are not required to set a registry secret on individually for all the apps they manage.
